nengo_spinnaker/simulator.py

```
@@ -148,11 +232,44 @@ def run(self, time_in_seconds=None, clean=True):
+                    # "Send signal 2 (meaning stop) to all executables with the
+                    #  app_id we've given them (usually 30)."
                     self.controller.txrx.app_calls.app_signal(
                         self.dao.app_id, 2)
             except Exception:
                 pass
 
+    def reset(self):
+        """Reset the Simulator.
+
+        The next simulation will start from the beginning.
+        """
+        # This is only really here to ensure that the behaviour is consistent
+        # with the reference simulator.  We currently don't allow multiple runs
+        # in sequence, so there must be a reset capability to allow reuse of
+        # the simulator.
+        self.executed = False
+
     def trange(self, dt=None):
-        dt = self.dt if dt is None else dt
-        return dt * np.arange(int(self.time_in_seconds/dt))
+        """Generate a list of time steps for the last simulation.
+
+        :returns: Numpy array of time steps.
+        """
+        if self.time_in_seconds is not None:
+            dt = self.dt if dt is None else dt
+            return dt * np.arange(int(self.time_in_seconds/dt))
+        else:
+            # XXX: This is horrible, but it beats getting an exception because
+            #      dividing None is invalid.  If the run time wasn't specified
+            #      probing would have been disabled.
+            # TODO: Allow some probing for unspecified run time... Will require
+            #       writing the final run time back somehow. (When we have
+            #       masses of bandwidth we could even probe on
+            #       on host)
+            #       The other option is to dynamically switch the model so that
+            #       we probe as best as we can on host when no run time is
+            #       specified.  This would require some rearranging of the
+            #       Builder.
+            return np.array([])
```

Maybe raise NotImplemented or some such thing here rather than silently letting things roll along (assuming the generator idea doesn't apply?)